src/lib/falcor/FalcorController.js

```diff
@@ -103,15 +101,65 @@ export default class FalcorController extends BaseComponent {
         });
       }
       else {
-        const err = new Error("FalcorPathSets: " + JSON.stringify(falcorPathSets) + " returned no data.")
-        console.error(err);
+        console.error("FalcorPathSets: " + JSON.stringify(falcorPathSets) + " returned no data.");
         this.safeSetState({
           ready: true,
           fetching: false,
           data: null,
-          error: err,
         });
       }
+      let globalError = getGlobalError();
+      if (globalError && globalError.message === "Response code 0") {
+        resetGlobalError();
+      }
+    }).catch((err) => {
+      try {
+        if (!(err instanceof Array)) {
+          if (!(err instanceof Error)) {
+            throw new Error("Unexpected Falcor Error. Caught error looks like this after JSON.stringify: " + JSON.stringify(err));
```

Comment:
Confusing. Why do you make a try, just to throw an error on purpose? Why not just build your string and just setGlobalError without this try stuff?